compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenOps.td

```diff
@@ -171,5 +171,54 @@ def IREECodegen_NullPointerOp :
   let assemblyFormat = "attr-dict";
 }
 
+//===----------------------------------------------------------------------===//
+// LoadFrom/StoreToMemref Ops
+//===----------------------------------------------------------------------===//
+
+def IREECodegen_LoadFromMemrefOp : Op<IREECodegen_Dialect, "load_from_memref",
+    [DeclareOpInterfaceMethods<MemoryEffectsOpInterface>]> {
+  let summary = [{loads a tensor from a memref}];
+  let description = [{
+    Loads a tensor from a memref with the same shape and element type. The
+    read_only attribute indicates that the source buffer from which the tensor
+    is read is a read only buffer. This hint can be used by bufferization to
+    determine whether or not the buffer that this op reads from may be written.
+  }];
+
+  let arguments = (ins
+    AnyStridedMemRef:$source,
+    UnitAttr:$read_only
+  );
+  let results = (outs
+    AnyRankedTensor:$result
+  );
+
+  let assemblyFormat = [{
+    $source attr-dict `:` type($source) `->` type($result)
+  }];
+
+  let hasVerifier = 1;
+}
+
+def IREECodegen_StoreToMemrefOp : Op<IREECodegen_Dialect, "store_to_memref",
+    [DeclareOpInterfaceMethods<MemoryEffectsOpInterface>]> {
+  let summary = [{stores a tensor into a memref}];
+  let description = [{
+    Stores a tensor into a memref with the same shape and element type.
```

Comment:
maybe replace "same" with "compatible" if we are going to allow compatible shapes.

```suggestion
    Stores a tensor into a memref with the compatible shape and the same element type.
```